features/dd-sdk-android-session-replay/src/test/kotlin/com/datadog/android/sessionreplay/internal/recorder/mapper/HiddenViewMapperTest.kt

```diff
@@ -0,0 +1,117 @@
+import com.datadog.android.sessionreplay.model.MobileSegment
+import com.datadog.android.sessionreplay.recorder.MappingContext
+import com.datadog.android.sessionreplay.recorder.SystemInformation
+import com.datadog.android.sessionreplay.utils.AsyncJobStatusCallback
+import com.datadog.android.sessionreplay.utils.GlobalBounds
+import com.datadog.android.sessionreplay.utils.ViewBoundsResolver
+import com.datadog.android.sessionreplay.utils.ViewIdentifierResolver
+import fr.xgouchet.elmyr.junit5.ForgeConfiguration
+import fr.xgouchet.elmyr.junit5.ForgeExtension
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.api.extension.Extensions
+import org.mockito.Mock
+import org.mockito.junit.jupiter.MockitoExtension
+import org.mockito.junit.jupiter.MockitoSettings
+import org.mockito.kotlin.any
+import org.mockito.kotlin.whenever
+import org.mockito.quality.Strictness
+
+@Extensions(
+    ExtendWith(MockitoExtension::class),
+    ExtendWith(ForgeExtension::class)
+)
+@MockitoSettings(strictness = Strictness.LENIENT)
+@ForgeConfiguration(ForgeConfigurator::class)
+internal class HiddenViewMapperTest {
+    @Mock
+    lateinit var mockViewIdentifierResolver: ViewIdentifierResolver
+
+    @Mock
+    lateinit var mockViewBoundsResolver: ViewBoundsResolver
+
+    @Mock
+    lateinit var mockAsyncJobStatusCallback: AsyncJobStatusCallback
+
+    @Mock
+    lateinit var mockMappingContext: MappingContext
+
+    @Mock
+    lateinit var mockSystemInformation: SystemInformation
+
+    @Mock
+    lateinit var mockGlobalBounds: GlobalBounds
+
+    @Mock
+    lateinit var mockInternalLogger: InternalLogger
+
+    private lateinit var testedViewMapper: HiddenViewMapper
+
+    @BeforeEach
+    fun setup() {
+        whenever(mockViewBoundsResolver.resolveViewGlobalBounds(any(), any()))
+            .thenReturn(mockGlobalBounds)
+
+        whenever(mockMappingContext.systemInformation)
+            .thenReturn(mockSystemInformation)
+
+        whenever(mockSystemInformation.screenDensity)
+            .thenReturn(1f)
+
+        testedViewMapper = HiddenViewMapper(
+            viewIdentifierResolver = mockViewIdentifierResolver,
+            viewBoundsResolver = mockViewBoundsResolver
+        )
+    }
+
+    @Test
+    fun `M return a placeholder with correct label W map()`(
+        @Mock mockView: View
+    ) {
+        // When
+        val wireframesList = testedViewMapper.map(
+            view = mockView,
+            asyncJobStatusCallback = mockAsyncJobStatusCallback,
+            internalLogger = mockInternalLogger,
+            mappingContext = mockMappingContext
+        )
+
+        val wireframe = wireframesList[0] as? MobileSegment.Wireframe.PlaceholderWireframe
+
+        // Then
+        checkNotNull(wireframe)
```

Comment:
we probably want to fail as early as possible if something goes wrong, and with this logic if `wireframesList[0]` is not `MobileSegment.Wireframe.PlaceholderWireframe` then we will get an error about the value being `null`, which is misleading.

maybe do something like this?

```suggestion

        // Then
        val wireframe = wireframesList[0]
        check(wireframe is MobileSegment.Wireframe.PlaceholderWireframe)
```